releasing.md

```diff
@@ -155,14 +155,46 @@ ks apply ${ENV} -c workflows
 Create a PR to update [jupyterhub_spawner.py](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/jupyterhub_spawner.py#L15) 
 to point to the newly built Jupyter notebook images.
 
-## Update the ksonnet configs
+## Create a release branch (if necessary)
 
-Create a PR to update the ksonnet configs to use the latest images built in the previous steps.
+If you aren't already working on a release branch (of the form `v${VER}-branch`, where `${VER}` is a major-minor version number), then create one.  Release branches serve several purposes:
 
-Presubmit tests should verify that the ksonnet configs work.
+1.  They allow a release wrangler or other developers to focus on a release without interrupting development on `master`,
+2.  they allow developers and sophisticated users to track the development of a release before a release candidate is declared, and
+3.  they simplify backporting critical bugfixes to a patchlevel release particular release stream (e.g., producing a `v0.1.1` from `v0.1-branch`), when appropriate.
```

Comment:
Probably worth mentioning the bit about providing stable ksonnet config users.

Reply:
Yes, that's what I was driving at with "they allow developers and sophisticated users to track the development of a release before a release candidate is declared" but I agree that having a ksonnet registry URL that always points to a relatively stable set of images in a particular release stream is worth calling out separately.